src/Adapter/MSTest.TestAdapter/TestingPlatformAdapter/MSTestBridgedTestFramework.cs

```diff
@@ -52,13 +54,50 @@ internal sealed class MSTestBridgedTestFramework : SynchronizedSingleSessionVSTe
 
         if (!MSTestDiscovererHelpers.InitializeDiscovery(request.AssemblyPaths, request.RunContext, request.FrameworkHandle))
         {
-            return Task.CompletedTask;
+            return;
         }
 
-#pragma warning disable CA1859 // Use concrete types when possible for improved performance
-        ITestExecutor testExecutor = new MSTestExecutor();
-#pragma warning restore CA1859 // Use concrete types when possible for improved performance
+        MSTestExecutor testExecutor = new();
         using (cancellationToken.Register(testExecutor.Cancel))
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                ApartmentState currentApartmentState = Thread.CurrentThread.GetApartmentState();
+                ApartmentState? requestedApartmentState = MSTestSettings.RunConfigurationSettings.ExecutionApartmentState;
+                if (requestedApartmentState is not null && currentApartmentState != requestedApartmentState)
+                {
+                    Thread entryPointThread = new(new ThreadStart(() => RunTests(testExecutor, request)))
+                    {
+                        Name = "MSTest Entry Point",
+                    };
+
+                    entryPointThread.SetApartmentState(requestedApartmentState.Value);
+                    entryPointThread.Start();
+
+                    try
+                    {
+                        var threadTask = Task.Run(entryPointThread.Join, cancellationToken);
+#if NET6_0_OR_GREATER
+                        await threadTask.WaitAsync(cancellationToken);
+#else
+#pragma warning disable VSTHRD103 // Call async methods when in an async method
+                        threadTask.Wait(cancellationToken);
+#pragma warning restore VSTHRD103 // Call async methods when in an async method
+#endif
+                    }
+                    catch
```

Comment:
@MarcoRossignoli I am wondering if that wouldn't be better to let the exception bubble up to the bridge/platform, WDYT?